app/wasm/test/custom_query_test.go

```diff
@@ -173,6 +173,96 @@ func TestQuerySpotPrice(t *testing.T) {
 	require.InEpsilonf(t, expected+swapFee, price, epsilon, fmt.Sprintf("Outside of tolerance (%f)", epsilon))
 }
 
+func TestQueryEstimatePrice(t *testing.T) {
+	actor := RandomAccountAddress()
+	osmosis, ctx := SetupCustomApp(t, actor)
+	epsilon := 1. // high epsilon (because of rounding?)
+
+	fundAccount(t, ctx, osmosis, actor, defaultFunds)
+
+	poolFunds := []sdk.Coin{
+		sdk.NewInt64Coin("uosmo", 12000000),
+		sdk.NewInt64Coin("ustar", 240000000),
+	}
+	// 20 star to 1 osmo
+	starPool := preparePool(t, ctx, osmosis, actor, poolFunds)
+
+	reflect := instantiateReflectContract(t, ctx, osmosis, actor)
+	require.NotEmpty(t, reflect)
+
+	// The contract/sender needs to have funds for estimating the price
+	fundAccount(t, ctx, osmosis, reflect, defaultFunds)
+
+	// Estimate swap rate
+	uosmo, err := poolFunds[0].Amount.ToDec().Float64()
+	require.NoError(t, err)
+	ustar, err := poolFunds[1].Amount.ToDec().Float64()
+	require.NoError(t, err)
+	swapRate := ustar / uosmo
+
+	// Query estimate cost (Exact in. No route)
+	amountIn := sdk.NewInt(1)
```

Comment:
Why 1?
No one ever swaps 0.000001 OSMO - less than gas fees.

Let's use normal numbers like 1.000.000 and this also limits the issues with rounding errors